tinygrad/runtime/metal.py

```diff
@@ -19,11 +20,8 @@ def __init__(self, shape): raise NotImplementedError("Metal runtime doesn't supp
 class CLBuffer:
   def __init__(self, size): self._cl = device.newBufferWithLength_options_(size, Metal.MTLResourceStorageModeShared)
   def copyin(self, b:np.ndarray):
-    # TODO: don't reallocate buffer!
-    self._cl = device.newBufferWithBytes_length_options_(
-      b.astype(np.float32).data,
-      b.size*4,
-      Metal.MTLResourceStorageModeShared)
+    wrapper = np.frombuffer(self._cl.contents().as_buffer(self._cl.length()), dtype=np.float32)
+    np.copyto(wrapper, b.reshape(wrapper.shape))
```

Comment:
Wait this really works?!? Nothing makes copies in wrapper?

Reply:
ended up switching to makeBuffer with no copy, wraps the memoryview from numpy ends up not allocating anything